src/test/java/uk/gov/hmcts/reform/orgrolemapping/domain/service/DroolBase.java

```diff
@@ -0,0 +1,70 @@
+package uk.gov.hmcts.reform.orgrolemapping.domain.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.kie.api.KieServices;
+import org.kie.api.command.Command;
+import org.kie.api.runtime.ExecutionResults;
+import org.kie.api.runtime.KieContainer;
+import org.kie.api.runtime.StatelessKieSession;
+import org.kie.internal.command.CommandFactory;
+import org.mockito.Mock;
+import uk.gov.hmcts.reform.orgrolemapping.domain.model.UserAccessProfile;
+import uk.gov.hmcts.reform.orgrolemapping.helper.TestDataBuilder;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.Map;
+
+import static org.mockito.Mockito.mock;
+import static uk.gov.hmcts.reform.orgrolemapping.domain.service.RequestMappingService.ROLE_ASSIGNMENTS_QUERY_NAME;
+import static uk.gov.hmcts.reform.orgrolemapping.domain.service.RequestMappingService.ROLE_ASSIGNMENTS_RESULTS_KEY;
+
+public abstract class DroolBase {
+
+    StatelessKieSession kieSession;
+    Map<String, Set<UserAccessProfile>> usersAccessProfiles;
+    List<Command<?>> commands;
+    ExecutionResults results;
+    Set<UserAccessProfile> allProfiles;
+
+    @Mock
+    private RetrieveDataService retrieveDataService = mock(RetrieveDataService.class);
+
+    @BeforeEach
+    public void setUp() {
+
+        usersAccessProfiles = TestDataBuilder.buildUserAccessProfileMap(false, false);
+
+        // Combine all the user profiles into a single collection for the rules engine.
+        allProfiles = new HashSet<>();
+        usersAccessProfiles.forEach((k, v) -> allProfiles.addAll(v));
+
+
+        // Set up the rule engine for validation.
+        KieServices ks = KieServices.Factory.get();
+        KieContainer kieContainer = ks.getKieClasspathContainer();
+        this.kieSession = kieContainer.newStatelessKieSession("org-role-mapping-validation-session");
+        this.kieSession.setGlobal("DATA_SERVICE", retrieveDataService);
```

Comment:
We are not using retrieveDataService to fetch any Case in ORM . Yo can remove it.

Reply:
Removed.